### ISSUE TITLE
Port akka-http to Scala.js

### ISSUE THREAD

**Issue by Atry** [unicredit/akka.js#55]
Though I will not expect running a HTTP server on Scala.js, some data structures from akka-http like `Path` and `Uri` are useful in Scala.js .

Another usage of akka-http.js is creating an in-memory mock server to handler fake HTTP requests (for testing purpose).

**Comment by andreaTP** [comment]
An Http server is totally possible if you run for example on Node, haven't looked in details how much akka-http is tighten to Jvm.
In this moment unfortunately I have few time, so PR are welcome!   :-)

**Comment by LogicalTime** [comment]
I agree, thought I am guessing it should be separate like Akka-Http is? I also long for those Uri and Path classes.

**Comment by andreaTP** [comment]
I guess Uri and Path are easily portable...
Since here we target just akka core, yes Akka-http port should live in a separate repo I belive

**Comment by LogicalTime** [comment]
There is now Scala-uri library that is is cross built. They has taken most of the pressure off this issue for me.
